service.py

```diff
@@ -60,6 +70,10 @@ def showErrorNotification(message):
 def getParams():
     result = {}
     paramstring = sys.argv[2]
+
+    if paramstring == '':
+        paramstring = "?" + get_user_input()
```

Comment:
I really don't know this but for what reason is the `?` needed? May maybe something like the following would be more direct
```python 
if not paramstring:
  result['url'] = paramstring[1:]
else:
  result['url'] = get_local_user_input()
  

```

Reply:
I was thinking that in future we might need to give option to user to select video quality, or pass any other customer parameters. That's why I did not pass URL directly in url variable. Kindly suggest.

Reply:
I would even then suggest to explicitly return the result dict already correctly assigned to the keys instead of parsing it from a string.

Forcing a quality via the resolver will not really be possible anymore. Basically all streams are now fragmented with a manifest. Many still provide one legacy all in one stream where there is not much to choose from. For Kodi you can set the desired quality of a fragmented stream within the adaptive input stream addon. We partially supported manifest since the last PR we were testing so much.